.github/workflows/R-CMD-check.yaml

```
@@ -42,10 +47,22 @@ jobs:
           use-public-rspm: true
 
       - if: runner.os == 'Linux'
-        run: sudo apt-get install -y protobuf-compiler libprotobuf-dev libprotoc-dev
+        run: sudo apt-get install -y protobuf-compiler libprotobuf-dev libprotoc-dev cmake libcurl4-openssl-dev libssl-dev
 
       - uses: r-lib/actions/setup-r-dependencies@v1
         with:
           extra-packages: rcmdcheck
+          path: substrait-r
+
+      - name: Checkout Arrow repo
+        uses: actions/checkout@v2
+        with:
+          repository: apache/arrow
+          path: arrow
+
+      - name: Install Arrow with ARROW_SUBSTRAIT turned on
+        run: .github/scripts/install_arrow_substrait.sh
```

```suggestion
        if: steps.cache.outputs.cache-hit != 'true'
        run: .github/scripts/install_arrow_substrait.sh
```
And this should skip compilation if cache was restored.